lib/verify.js

```diff
@@ -1,16 +1,32 @@
 // @ts-check
 
+const SemanticReleaseError = require('@semantic-release/error');
 const verifyPkg = require('./verify-pkg');
 const verifyAuth = require('./verify-auth');
 const verifyOvsxAuth = require('./verify-ovsx-auth');
 const verifyTarget = require('./verify-target');
+const { isOvsxEnabled } = require('./utils');
 
 module.exports = async (pluginConfig, { logger, cwd }) => {
   await verifyPkg();
   await verifyTarget();
 
   if (pluginConfig?.publish !== false) {
-    await verifyAuth(logger, cwd);
+    try {
+      await verifyAuth(logger, cwd);
+    } catch (err) {
+      if (
+        err instanceof SemanticReleaseError &&
+        err.code === 'ENOVSCEPAT' &&
+        isOvsxEnabled()
+      ) {
+        logger.log(
+          'The vsce personal access token is missing (set the `VSCE_PAT` environment variable) but publish to OpenVSX is enabled.',
+        );
+      } else {
+        throw err;
+      }
+    }
```

Comment:
Instead of trying and erroring, you can avoid calling `verifyVsceAuth` or `verifyOvsxAuth` if they are not enabled.

You can do the verification for the presence of one or another here (the one you added to `publish` and I asked to migrate to here).

Something like:

```
if (pluginConfig.publish || pluginConfig.publish === 'undefined') {
  // check if at least one environment variable is present (vsce or ovsx)

  // if vscePublishEnabled, verifyVsceAuth

  // if ovsxPublishEnabled, verifyOvsxAuth
}
```

Reply:
please see 79c40e0fc1ded0a2d806d8d30af2b72beb61cbec